hil-test/tests/gpio.rs

```diff
@@ -80,6 +88,49 @@ mod tests {
         ctx
     }
 
+    #[test]
+    async fn test_async_edge(ctx: Context) {
+        let counter = AtomicUsize::new(0);
+        let Context {
+            mut io2, mut io4, ..
+        } = ctx;
+        embassy_futures::select::select(
+            async {
+                loop {
+                    io2.wait_for_rising_edge().await.unwrap();
+                    counter.fetch_add(1, Ordering::SeqCst);
+                }
+            },
+            async {
+                for _ in 0..5 {
+                    io4.set_high().unwrap();
+                    Timer::after(Duration::from_millis(25)).await;
+                    io4.set_low().unwrap();
+                    Timer::after(Duration::from_millis(25)).await;
+                }
+            },
+        )
+        .await;
+        assert_eq!(counter.load(Ordering::SeqCst), 5);
+    }
+
+    #[test]
+    async fn test_pins_can_wait(_ctx: Context) {
+        let mut first = unsafe { GpioPin::<Unknown, 0>::steal() }.into_pull_down_input();
+        // convenient way to get the last pin without cfgs for all chips
+        let mut last =
+            unsafe { GpioPin::<Unknown, { (NUM_PINS - 1) as u8 }>::steal() }.into_pull_down_input();
```

Comment:
Depending on the test's intent: since NUM_PINS was actually one less for some of the chips, relying on that constant here isn't a good regression test - changing the constant back will still result in the test passing, even though the last pin would no longer be usable.

Reply:
Good point! I've updated the test. Thanks!